### PR TITLE
Added a TransactionFeeRule for the x42 network.

### PR DESCRIPTION
**Pull request by DennisAMenace**

### CODE REVIEW

File: src/Networks/x42/x42/Networks/x42Test.cs [hunk 1]
```diff
@@ -73,6 +74,13 @@ public x42Test()
                 [BuriedDeployments.BIP66] = 0
             };
 
+            var bip9Deployments = new x42BIP9Deployments
+            {
+                [x42BIP9Deployments.ColdStaking] = new BIP9DeploymentsParameters("ColdStaking", 27, BIP9DeploymentsParameters.AlwaysActive, 999999999, BIP9DeploymentsParameters.DefaultMainnetThreshold),
```
Comment (dangershony) [inline src/Networks/x42/x42/Networks/x42Test.cs:79]: Was this activated on on x42 already?

File: src/Features/Blockcore.Features.Miner/BlockDefinition.cs [hunk 2]
```diff
@@ -410,7 +410,9 @@ protected virtual void AddTransactions(out int nPackagesSelected, out int nDesce
                     packageSigOpsCost = modit.SigOpCostWithAncestors;
                 }
 
-                if (packageFees < this.BlockMinFeeRate.GetFee((int)packageSize))
```
Comment (dangershony) [inline src/Features/Blockcore.Features.Miner/BlockDefinition.cs:413]: if you set BlockMinFeeRate to zero doe it not have the same effect?

File: src/Networks/x42/x42/Networks/x42Test.cs (hunk 1 above)
Comment (DennisAMenace) [reply src/Networks/x42/x42/Networks/x42Test.cs:79]: Yes, this is for the testnet.

File: src/Features/Blockcore.Features.Miner/BlockDefinition.cs (hunk 2 above)
Comment (DennisAMenace) [reply src/Features/Blockcore.Features.Miner/BlockDefinition.cs:413]: Yes, however, If the OP_RETURN data exists, we still want a fee, so if you look at the x42 override, I have the check for it there..

Comment (dangershony) [reply src/Features/Blockcore.Features.Miner/BlockDefinition.cs:413]: Then just make a rule for x42 (and a mempool rule) and it wills till work no?

Comment (DennisAMenace) [reply src/Features/Blockcore.Features.Miner/BlockDefinition.cs:413]: I don't see how that would work, as other chains need the default behavior.

**Comment by DennisAMenace** [comment]
Closing this in favor of another approach.

**Pull request closed without merging**